### ISSUE TITLE
decoding to output (dev/test) - 98% of CPU loading? 

### ISSUE THREAD

**Issue by myurushkin** [tensorflow/nmt#110]
Hi there!

My GPU: gtx1070,
TF 1.3 

Is it ok that evaluation of test/dev datasets is performed on CPU mostly?
GPU loading: 20%
CPU loading: >90%

In my case training doesn't take much time in comparision with evaluation. Is it ok?

Best regards,
Mikhail

**Comment by oahziur** [comment]
@myurushkin Are you using length penalty weight > 0.0 for beam search? You may disable it in your training for now as we found it doesn't utilize GPU well. A fix will be pushed to the future version of tensorflow.

**Comment by myurushkin** [comment]
Yep, I have used one of the default configurations: wmt16_gnmt_4_layer.json
length_penalty_weight=1 in this case. I'll set this parameter to 0.
Thanks!

**Comment by okuchaiev** [comment]
@oahziur is there a TF issue to track this? Thanks!

**Comment by oahziur** [comment]
@okuchaiev This should be fixed in tensorflow version 1.4
